cmd/gapit/unpack.go

```diff
@@ -111,3 +134,74 @@ func (u *unpacker) msgString(msg proto.Message) string {
 	}
 	return str
 }
+
+type dumper struct {
+	ids      map[uint64]bool
+	children bool
+	dir      string
+	found    int
+	count    int
+}
+
+func (u *dumper) BeginGroup(ctx context.Context, msg proto.Message, id uint64) error {
+	if _, ok := u.ids[id]; ok {
+		u.found = 1
```

Comment:
I believe groups can overlap in the stream, in which case i think this approach may break.

Reply:
This is correct, we have this in Vulkan in a few places where we have API calls that block. We mark the start and end completely separately. You may have to buffer the begin, until you see the end, then print it out all at once.